test/form/samples/tdz-common/main.js

```diff
@@ -0,0 +1,39 @@
+console.log(function() {
+	if (x) return "HELLO"; // TDZ
+	const x = 1;           // keep
+	return "WORLD";        // not reached
```

Comment:
Although dropping the`return "WORLD";` line for this test doesn't affect the outcome of the preserved TDZ violation above, I am not sure why it was dropped due to a side effect being reported on the Identifier `x` in `if (x) return "HELLO";`. If this example were changed to use `var` instead of `const`, the rollup result in this PR would be incorrect due to the dropped `return "WORLD";`:

Expected:
```
$ echo 'console.log(function(){if(x)return"HELLO";var x=1;return"WORLD"}());' | node
WORLD
```
Actual using this PR:
```
$ echo 'console.log(function(){if(x)return"HELLO";var x=1;return"WORLD"}());' | rollup --silent
console.log(function(){if(x)return "HELLO";var x=1;}());
```
```
$ echo 'console.log(function(){if(x)return"HELLO";var x=1;return"WORLD"}());' | rollup --silent | node
undefined
```
Note that rollup v2.50.6 is also wrong, but in a different way:
```
$ echo 'console.log(function(){if(x)return"HELLO";var x=1;return"WORLD"}());' | node_modules/.bin/rollup --silent
console.log(function(){return "HELLO";}());
```